src/stream/mod.rs

```diff
@@ -243,16 +247,90 @@ impl<R: Read + Seek> Reader<R> {
     }
 
     fn is_binary(reader: &mut R) -> Result<bool, Error> {
-        fn from_io_offset_0(err: io::Error) -> Error {
-            ErrorKind::Io(err).with_byte_offset(0)
+        Self::rewind(reader)?;
+        let is_binary = Self::reader_matches(reader, b"bplist00")?;
+        Self::rewind(reader)?;
+        Ok(is_binary)
+    }
+
+    fn is_xml(reader: &mut R) -> Result<bool, Error> {
+        const UTF32_BE_BOM: [u8; 4] = [0, 0, 0xfe, 0xff];
```

Comment:
`is_xml` would be accurately be described as `should_use_xml_reader`. The ASCII reader shouldn't be given a document with a BOM, and quick-xml will validate support for whatever BOM it's given. I've includeded BOMs I know quick-xml doesn't support just so it's forwards compatible and so that the ASCII reader is still avoided if/when those other BOMs are parsed.

We could add support for GNUstep plists at a later date, which supports UTF-8, without an API break.